File: src/main/java/com/multicampus/gamesungcoding/a11ymarketserver/feature/seller/dto/SellerProductUpdateRequest.java
package com.multicampus.gamesungcoding.a11ymarketserver.feature.seller.dto;

import com.multicampus.gamesungcoding.a11ymarketserver.feature.product.dto.ImageMetadata;
import com.multicampus.gamesungcoding.a11ymarketserver.feature.product.entity.ProductStatus;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;

import java.util.List;

public record SellerProductUpdateRequest(
        @NotBlank
        String productName,

        @NotBlank
        String productDescription,

        @NotBlank
        String categoryId,

        @NotNull
        @Min(0)
        Integer productPrice,

        @NotNull
        @Min(0)
        Integer productStock,

        ProductStatus productStatus,

        List<ImageMetadata> imageMetadataList) {
}

File: src/main/kotlin/com/multicampus/gamesungcoding/a11ymarketserver/feature/seller/dto/SellerProductUpdateRequest.kt
package com.multicampus.gamesungcoding.a11ymarketserver.feature.seller.dto

import com.multicampus.gamesungcoding.a11ymarketserver.feature.product.dto.ImageMetadata
import com.multicampus.gamesungcoding.a11ymarketserver.feature.product.entity.ProductStatus
import jakarta.validation.constraints.Min
import jakarta.validation.constraints.NotBlank

data class SellerProductUpdateRequest(
    @field:NotBlank val productName: String,
    @field:NotBlank val productDescription: String,
    @field:NotBlank val categoryId: String,
    @field:Min(0) val productPrice: Int,
    @field:Min(0) val productStock: Int,
    val productStatus: ProductStatus,
    val imageMetadataList: List<ImageMetadata>? = null
)
